crates/cxx-qt-lib-extras/src/gui/qapplication.rs

```diff
@@ -19,11 +21,26 @@ mod ffi {
         type QVector_QByteArray = cxx_qt_lib::QVector<QByteArray>;
         include!("cxx-qt-lib/qfont.h");
         type QFont = cxx_qt_lib::QFont;
+        include!("cxx-qt-lib/qguiapplication.h");
+        type QGuiApplication = cxx_qt_lib::QGuiApplication;
 
         include!("cxx-qt-lib-extras/qapplication.h");
         type QApplication;
     }
 
+    #[namespace = "rust::cxxqt1"]
+    unsafe extern "C++" {
+        include!("cxx-qt/casting.h");
+
+        #[doc(hidden)]
+        #[rust_name = "upcast_qapplication"]
+        unsafe fn upcastPtr(thiz: *const QApplication) -> *const QGuiApplication;
+
+        #[doc(hidden)]
+        #[rust_name = "downcast_qapplication"]
+        unsafe fn downcastPtr(base: *const QGuiApplication) -> *const QApplication;
+    }
```

Comment:
This will not be required once #1202 lands as automatic up/down casting implementations will be done via a C++Qt block, such as

```rust
unsafe extern "C++Qt" {
   #[base = "QGuiApplication"]
   type QApplication;
}
```